ucan/src/tests/builder.rs

```diff
@@ -132,6 +138,57 @@ async fn it_prevents_duplicate_proofs() {
 
     assert_eq!(
         next_ucan.proofs(),
-        &vec![Cid::try_from(ucan).unwrap().to_string()]
+        &vec![ucan
+            .to_cid(UcanBuilder::<PatchedKeyPair>::default_hasher())
+            .unwrap()
+            .to_string()]
     )
 }
+
+#[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
+#[cfg_attr(not(target_arch = "wasm32"), tokio::test)]
+pub async fn it_can_use_custom_hasher() {
+    let identities = Identities::new().await;
+    let mut did_parser = DidParser::new(SUPPORTED_KEYS);
+
+    let leaf_ucan = UcanBuilder::default()
+        .with_hasher(Code::Blake2b256)
```

Comment:
I think it might make more sense for the hasher to be passed as an optional second parameter where it is used e.g., `witnessed_by(ucan, Some(Code::Blake2b256))`.

Rationale:

As written, the hasher is a stateful field of the builder. But, being able to change the hasher with a chained call implies that we may want to use different hashes for different proofs.

With that use case in mind, it makes more sense to pass the desired hash in with the proof. OTOH if that use case is out of scope, I would say we shouldn't allow the hasher to be changed after initializing the builder (since the hash used will change depending on the call order of the methods, and that would be confusing / surprising).

Reply:
I think the `witnessed_by(ucan, Some(Code::Blake2b256))` approach is the best for now. An API change however, but rather straight forward, and less stateful than the `with_hasher` approach

Reply:
or: `witnessed_by_with_hasher(&Ucan, Code)` for non-default hashing proofs in addition to e.g. `witnessed_by(&Ucan)`

Reply:
> I think the `witnessed_by(ucan, Some(Code::Blake2b256))` approach is the best for now. An API change however, but rather straight forward, and less stateful than the `with_hasher` approach

Pushed a take on this